providers/fips.module.sources

```diff
@@ -22,12 +23,14 @@ crypto/aes/asm/aesp8-ppc.pl
 crypto/aes/asm/aest4-sparcv9.pl
 crypto/aes/asm/aesv8-armx.pl
 crypto/aes/asm/bsaes-armv7.pl
+crypto/aes/asm/bsaes-armv8.S
 crypto/aes/asm/bsaes-x86_64.pl
 crypto/aes/asm/vpaes-armv8.pl
 crypto/aes/asm/vpaes-ppc.pl
 crypto/aes/asm/vpaes-x86.pl
 crypto/aes/asm/vpaes-x86_64.pl
 crypto/asn1_dsa.c
+crypto/bn/asm/../bn_local.h
```

Comment:
duplicate due to .. :)

Reply:
Not sure how to remove these.  It's not a problem, having the same file more than once isn't going to generate false results.

Reply:
I'd much prefer to have duplicates than risk missing something.

Reply:
Er...  So with these "aliases", `providers/fips-sources.checksums.new` will contain two lines:

```
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  crypto/bn/asm/../bn_local.h
...
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  crypto/bn/asm/bn_local.h
```

while if you don't have them, you'll only have the one line:

```
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  crypto/bn/asm/bn_local.h
```

You don't think that will affect the total checksum in `providers/fips.checksum.new`?

Unfortunately, File::Spec doesn't help resolving these kinds of paths, not even File::Spec->canonpath().  This is documented.  From `man File::Spec`:

> canonpath
> No physical check on the filesystem, but a logical cleanup of a path.
> 
>     $cpath = File::Spec->canonpath( $path ) ;
> Note that this does *not* collapse _x/../y_ sections into _y_.  This is
> by design.  If _/foo_ on your system is a symlink to _/bar/baz_, then
> _/foo/../quux_ is actually _/bar/quux_, not _/quux_ as a naive _../_-removal
> would give you.  If you want to do this kind of processing, you
> probably want "Cwd"'s "realpath()" function to actually traverse the
> filesystem cleaning up paths like this.

However, since *we* know that there are no symlinks (or shouldn't be), *we* can choose to clean up paths like that.  I believe that the trick to do so is `File::Spec->abs2rel(Cwd::realpath($path))`

Reply:
Of course the change will impact the checksum.  What I meant is that it doesn't matter having a duplicate like this so long as it is consistently present.  If one line changes, the other does too.  The overall checksum changes regardless.

Reply:
`s:/[^/]*/../:/:g` comes close.  Close enough that it handles all current cases.

Reply:
I've pushed a change that gets rid of these `..` cases.